utils/data_cleaning.py

```
@@ -1,37 +1,368 @@
+- Robust numeric cleaning with comprehensive error handling
+- Outlier detection and handling
+- Data type validation and conversion
+- Data quality validation
+- Input validation and edge case handling
+- Informative logging and error messages
+- Support for various data formats
+
 Driven by central configuration from config.py
 """
+import warnings
+from typing import Any, Dict, List, Optional, Tuple, Union
```

Ruff (0.14.13)

`typing.Dict` is deprecated, use `dict` instead

(UP035)